src/debugger.ts

```diff
@@ -165,13 +163,14 @@ export class Debugger
     let initialMessage = "";
     let initialized = false;
 
-    const workspaceFolder = session.workspaceFolder;
-    if (!workspaceFolder) {
-      throw new Error("Debugging requires a workspace folder to be opened");
+    const cwd = session.workspaceFolder?.uri.fsPath;
+    let workspaceName = session.workspaceFolder?.name;
+
+    if (!workspaceName) {
+      workspaceName = cwd ? path.basename(cwd) : "unknown";
```

Comment:
Should we error out when the workspace name is unknown? Is it possible to enter this case but still can use the debugger correctly 🤔

Reply:
That's a good point. I started saving our resolved workspace in the debug session configuration instead, so that we can reuse it when spawning the debugger.

That way, the error handling happens when resolving debug configurations and spawning just uses the information saved directly.